src/main.c

```diff
@@ -436,6 +442,17 @@ void on_keyboard_event(struct keyboard_collection* collection,
 	// TODO handle multiple symbols
 	xkb_keysym_t symbol = xkb_state_key_get_one_sym(keyboard->state, key);
 
+	if (symbol == XKB_KEY_F12) {
+		if (!is_pressed) {
+			inhibitor_toggle(inhibitor);
+		}
+		return;
+	}
+
+	if (!inhibitor_is_inhibited(inhibitor)){
```

Comment:
Why not pass keyboard events when the input isn't inhibited?

Reply:
In some cases it is not something we want. In my case, I have a shortcut that is triggered when <super> is released, making it possible to launch rofi with a single key press, but if the client receives input when the inhibitor is not active it will receive the <super> key press might trigger something on the client side (in this case, windows open the start menu) 
The logic is still not really correct though, I fixed it in the update